src/components/toggle/toggle.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import {findDOMNode} from 'react-dom';
+import {SBComponent} from 'stylable-react-component';
+import style from './toggle.st.css';
+
+export interface Props {
+    checked?: boolean,
+    error?: boolean,
+    disabled?: boolean,
+    onChange?: (selected: boolean) => void,
+    label?: string,
+    tabIndex?: number,
+    rtl?: boolean
+}
+export interface State {
+    focus: boolean,
+    click: boolean
+}
+
+@SBComponent(style)
+export default class Toggle extends React.Component<Props, State> {
+    static defaultProps = {
+        checked: false,
+        disabled: false,
+        error: false,
+        rtl: false
+    }
+    state = {
+        focus: false,
+        click: false
```

Comment:
agree. That was rather hack. Also `click` used as flag and has nothing about render. Moved this into `this.shouldResetFocus` property